libs/api/metadata-converter/src/lib/iso19139/read-parts.ts

```diff
@@ -907,16 +908,12 @@ export function readTemporalExtents(rootEl: XmlElement) {
 }
 
 export function readSpatialExtents(rootEl: XmlElement) {
-  const extractGeometries = (rootEl: XmlElement): Geometry[] => {
+  const extractGeometry = (rootEl: XmlElement): Polygon | MultiPolygon => {
     if (!rootEl) return null
     return pipe(
-      findChildrenElement('gmd:polygon', false),
-      mapArray((el) => {
-        const elements = el.children.filter(
-          (child) => child instanceof XmlElement
-        )
-        return readGeometry(elements[0] as XmlElement)
-      })
+      findChildElement('gmd:polygon', false),
+      firstChildElement,
+      map((el) => readGeometry(el) as Polygon | MultiPolygon)
```

Comment:
@jahow Can you explain why this cast to `Polygon | MultiPolygon`, when the GML can be of any kind, and OL is parsing it correctly?

Reply:
sorry this can be removed, it is indeed useless!